src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
It is not readable code, when you have such long conditions it is better to create variable like `isZeroSide = a <= 0 || b <= 0 || c <= 0` `isSizePossible = a + b <= c || a + c <= b || b + c <= a` and then you can to make two error messages, `must have a positive number` and `one of the sides is too long`